### PR TITLE
Refactor OTLP exporter env config to be shared across all exporters

### PR DESCRIPTION
**Pull request by dmathieu**
The envconfig from both OTLP exporters is highly repeated code.
As we want to remove the gRPC dependency from HTTP packages, we would need to duplicate even more.

So this is a proposal to make it somewhat reusable.
There is still some code repetition, as it wouldn't be trivial to extract `GenericOption` into a shared package. However, I believe this moves in the right direction, as quite a bit of logic is now being shared.

### CODE REVIEW

**Comment by dmathieu** [comment]
This doesn't introduce any changes to public APIs. So I didn't add a changelog entry.

**Comment by pellared** [comment]
I just made a quick look and the approach LGTM.

**Comment by Aneurysm9** [comment]
> This doesn't introduce any changes to public APIs. So I didn't add a changelog entry.

I still think this should have a changelog entry.  It may not change any public APIs, but it may change behavior and having a changelog entry may help users to know that there was a change in this area that may be related to any issues they experience.

Skipping the changelog entry should be for things like documentation changes or test-only changes.

**Comment by pellared** [comment]
> it may change behavior 

@dmathieu Is it so? Could you please explain in the PR description? It looks like pure refactoring which should be invisible to the user.

**Comment by dmathieu** [comment]
No, this shouldn't change the behavior in any way a user of the exporter would notice. The API remains the same, and environment variables also keep the same behavior.
There can always be unseen regressions obviously.

**Comment by pellared** [comment]
> Skipping the changelog entry should be for things like documentation changes or test-only changes.

@Aneurysm9 As a user, I find internal refactoring changelog entries as noise. Still, up to you. I respect your decisions 👍

**Comment by Aneurysm9** [comment]
> this shouldn't change the behavior in any way a user of the exporter would notice. The API remains the same, and environment variables also keep the same behavior.
> There can always be unseen regressions obviously.

This is the core of my point.  While no behavior change is intended, that is different from no behavior change being possible.  Without an indication in the changelog that a change was made to this exporter a user who is experiencing a regression may be misled and delayed in identifying this change as a potential cause.

File: exporters/otlp/internal/envconfig/envconfig.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+// Apply runs every ConfigFn
+func (e *EnvOptionsReader) Apply(opts ...ConfigFn) {
+	for _, o := range opts {
+		o(e)
+	}
+}
+
+// GetEnvValue gets an OTLP environment variable value of the specified key using the GetEnv function.
```
Comment (hanyuancheung) [inline exporters/otlp/internal/envconfig/envconfig.go:45]:
```suggestion
// GetEnvValue gets an OTLP environment variable value of the specified key 
// using the GetEnv function.
```

**Review by hanyuancheung — APPROVED** [review]
LGTM

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**